src/wayland/dmabuf/mod.rs

```diff
@@ -0,0 +1,465 @@
+//!         // both client-side and server side.
+//!         // You can now free the associated resource in your renderer.
+//!     }
+//! }
+//!
+//! impl DmabufHandler for MyDmabufHandler {
+//!     type BufferData = MyBufferData;
+//!     fn validate_dmabuf(&mut self, info: BufferInfo) -> Result<Self::BufferData, ()> {
+//!         /* validate the dmabuf and import it into your renderer state */
+//!         Ok(MyBufferData { /* ... */ })
+//!     }
+//! }
+//!
+//! // Once this is defined, you can in your setup initialize the dmabuf global:
+//!
+//! # fn main() {
+//! # let mut event_loop = wayland_server::calloop::EventLoop::<()>::new().unwrap();
+//! # let mut display = wayland_server::Display::new(event_loop.handle());
+//! // define your supported formats
+//! let formats = vec![
+//!     /* ... */
+//! ];
+//! let dmabuf_global = init_dmabuf_global(
+//!     &mut display,
+//!     formats,
+//!     MyDmabufHandler,
+//!     None // we don't provide a logger in this example
+//! );
+//! # }
+//! ```
+
+use std::{cell::RefCell, os::unix::io::RawFd, rc::Rc};
+
+pub use wayland_protocols::unstable::linux_dmabuf::v1::server::zwp_linux_buffer_params_v1::Flags;
+use wayland_protocols::unstable::linux_dmabuf::v1::server::{
+    zwp_linux_buffer_params_v1::{
+        Error as ParamError, RequestHandler as ParamRequestHandler, ZwpLinuxBufferParamsV1 as BufferParams,
+    },
+    zwp_linux_dmabuf_v1,
+};
+use wayland_server::{protocol::wl_buffer, Display, Global, NewResource};
+
+/// Representation of a Dmabuf format, as advertized to the client
+pub struct Format {
+    /// The format identifier.
+    ///
+    /// It must be a `DRM_FORMAT` code, as defined by the libdrm's drm_fourcc.h. The Linux kernel's DRM
+    /// sub-system is the authoritative source on how the format codes should work.
+    pub format: u32,
+    /// The supported dmabuf layout modifier.
+    ///
+    /// This is an opaque token. Drivers use this token to express tiling, compression, etc. driver-specific
+    /// modifications to the base format defined by the DRM fourcc code.
+    pub modifier: u64,
+    /// Number of planes used by this format
+    pub plane_count: u32,
+}
+
+/// A plane send by the client
+pub struct Plane {
+    /// The file descriptor
+    pub fd: RawFd,
+    /// The plane index
+    pub plane_idx: u32,
+    /// Offset from the start of the Fd
+    pub offset: u32,
+    /// Stride for this plane
+    pub stride: u32,
+    /// Modifier for this plane
+    pub modifier: u64,
+}
+
+/// The complete information provided by the client to create a dmabuf buffer
+pub struct BufferInfo {
+    /// The submitted planes
+    pub planes: Vec<Plane>,
+    /// The width of this buffer
+    pub width: i32,
+    /// The height of this buffer
+    pub height: i32,
+    /// The format in use
+    pub format: u32,
+    /// The flags applied to it
+    ///
+    /// This is a bitflag, to be compared with the `Flags` enum reexported by this module.
```

Comment:
Isn't there a nice crate to handle bitmaps a little bit better?

Reply:
The correct thing to do would be to add the indication that this is a bitflag to the protocol file, so that wayland-scanner would directly generate it as a bitflag rather than an enum...